internal/docker/config.go

```diff
@@ -55,8 +57,61 @@ func (config Config) validate() error {
 	if config.Endpoint == "" {
 		return errors.New("config.Endpoint must be specified")
 	}
-	if config.DockerAPIVersion < minimalRequiredDockerAPIVersion {
-		return fmt.Errorf("Docker API version must be at least %v", minimalRequiredDockerAPIVersion)
+	if err := VersionIsValidAndGTE(config.DockerAPIVersion, minimumRequiredDockerAPIVersion); err != nil {
+		return err
+	}
+	return nil
+}
+
+type apiVersion struct {
+	major int
+	minor int
+}
+
+func NewAPIVersion(version string) (string, error) {
```

Comment:
Can we use "github.com/hashicorp/go-version" instead of implementing parsing and comparison ourselves? We already extensively rely on that module

Reply:
This is only parsing and comparison is done w/ the docker lib (`versions.LessThan()` w/o validation so we need some validation implementation). I can if you really want to but the hashicorp lib appears to generally be for semver, which the client api isn't.

Reply:
Do you mean that docker API can have a version not compatible with semver, so we cannot use parsing/validation from the hashicorp lib?

Reply:
Yes, the hashicorp version only supports three segment versions (`"1.40.0"` instead of `"1.40"`) so would need additional reconciliation (`fmt.Sprintf("%d.%d", ver.Segments()[0], ver.Segments()[1])`) to adopt the lib.

Reply:
Ok, sounds good 👍